api/lib/api_web/resolvers/domain.ex

```diff
@@ -4,11 +4,25 @@ defmodule ApiWeb.Resolvers.Domain do
   """
   alias Api.Clients
 
+  defp get_definitions(language, tokens) do
+    case Clients.LanguageService.definitions(language, tokens) do
+      {:ok, definitions} -> definitions
+      _ -> :error
+    end
+  end
+
   def get_words_in_text(_parent, %{language: language, text: text}, _resolution) do
     {:ok, words} = Clients.LanguageService.tag(language, text)
-    tokens = Enum.map(words, fn word -> Map.fetch!(word, :token) end)
-    {:ok, definitions} = Clients.LanguageService.definitions(language, tokens)
 
+    # Request definitions in batches of 10
+    # Too many words in a request will still cause problems
+    definitions =
+      Enum.map(words, fn word -> Map.fetch!(word, :token) end)
+      |> Enum.chunk_every(10)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Pipe chain should start with a raw value.](https://app.codacy.com/manual/lucaskjaero/foreign-language-reader/pullRequest?prid=5165490)